docs/README.md

```diff
@@ -0,0 +1,143 @@
+## Frontend and Backend Directories
+### Frontend
+The code for the frontend views and endpoints are in the [`/frontend`](/frontend) directory. These include things like error messages, user authentication state displays and user email input pages.
+
+### Backend
+The code for the backend endpoints are stored in the [`/api`](/api) directory. These include things like SSO login redirects and *magic-link* endpoints
+
+## Other directories
+
+- [`/models`](/models) - the shared models for the service (including models and lookups for objects on other services)
+- [`/security`](/security) - security utils used by both front and backend
+- [`/openapi`](/openapi) - the OpenApi config for the backend routes at [`/api`](/api)
+- [`/config`](/config) - the environment config files
+- [`/swagger`](/swagger) - the customised components of swagger (used by [`build.py`](/build.py) to build a customised version of swagger without the default search bar at the top)
+- [`/.github`](/.github) - GitHub deployment workflow files
+- [`/run`](/run) - Gunicorn worker config scripts for dev/test environments
+- [`/tests`](/tests) - tests
+- [`/docs`](/docs) - these docs
+
+## JWT Auth Principles
+### How it works
+This service issues *signed* tokens, like a digital ticket, which is saved in a user's browser as cookie. The token contains *claims* about the user (like their name, email and account id), and a hash of that data which is signed using the private key of an RSA256 key pair.
+
+When this token is saved in a cookie by this authenticator, with both this authenticator and the other microservices on a common domain like `.access-funding.levellingup.gov.uk` it can then be read by any other microservices.
+
+Those services can check the *signature* on the token using the public key from the RSA256 keypair to verify that the data in it has been issued by this authenticator service.
+
+For this to work, an RSA256 public/private key pair needs to be created, with at least the private key being available in this authenticator's environment as an env var to sign the token, and on any service that needs to be able to verify the signature, the public key of the pair needs to be available as a var to verify the signature.
+
+The benefit of this shared keypair with token approach is that the other services do not need to make a call to a central service to see if the current user is authenticated, they just need to check the signature is valid (and that it is not out of expired) to confirm that the user is who they say they are.
+
+We set an expiry time on the token to set the time after which the other services should consider the token invalid, and redirect the user back to this authenticator service to revalidate their ID and get issued with fresh token.
+
+### Where we use it
+We set our auth JWT cookie, start the user's session and redirect to their chosen service in one step in [AuthSessionView.create_session_and_redirect](/api/session/auth_session.py#L99).
```

Comment:
Could we use permalinks to revisions for these code links instead e.g. https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet so they won't rot over time?

Reply:
Yep totally - good suggestion - updated.